src/sentry/api/endpoints/project_searches.py

```diff
@@ -17,6 +17,8 @@ class SavedSearchSerializer(serializers.Serializer):
 
 
 class ProjectSearchesEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectSearchPermission,)
```

Comment:
@nimishsinghal we already have permissions for this though. We dont really want/need granular per-search permissions (which is what these would suggest).

We could do something like:

- project:read => can set their own default saved search
- project:write => can set default team saved search
- project:write => can remove saved searches

Now if we want to allow them to create *new* saved searches, we'd have to augment the data model, because we wouldn't really want those exposed to the team by default. IMO those are two diff concerns from defaulting a saved search and exposing member-only saved searches.

If we allowed member level saved searches, it'd be something along the lines of:

- an optional owner field on SavedSearch model (must be nullable)
- an option to "share this saved search with the team"
- members with project:write can share with team
- members with project:read can only create private searches
- private searches show up differently in the UI to suggest they're private